math/src/field/f64/mod.rs

```diff
@@ -569,3 +551,13 @@ fn mod_reduce(x: u128) -> u64 {
     let (result, over) = tmp0.overflowing_add(tmp1);
     result.wrapping_add(E * (over as u64))
 }
+
+/// Squares the base N number of times and multiplies the result by the tail value.
```

Comment:
So: logarithmic terms:      (logx << N) + logy.

(No need to use this as a comment, just confirming.)

Reply:
I think so.